tensorflow_probability/python/bijectors/masked_autoregressive.py

```diff
@@ -676,6 +679,50 @@ class AutoregressiveNetwork(tf.keras.layers.Layer):
   distribution.log_prob(np.ones((3, 2), dtype=np.float32))
   ```
 
+  The `conditional` argument can be used to instead build a conditional density.  
+  estimator. To do this the conditioning variable must be passed as a `kwarg`:
+
+  ```python
+  # Generate data as the mixture of two distributions.
+  n = 2000
+  c = np.r_[
+  np.zeros(n//2),
+  np.ones(n//2)
+  ]
+  x = np.r_[
+  np.random.randn(n//2).astype(dtype=np.float32) * 2,
+  np.random.randn(n//2).astype(dtype=np.float32) + 5
+  ]
+
+  # Density estimation with MADE.
+  made = tfb.AutoregressiveNetwork(params=2, hidden_units=[10, 10])
+
+  distribution = tfd.TransformedDistribution(
+    distribution=tfd.Sample(tfd.Normal(loc=0., scale=1.), sample_shape=[1]),
+    bijector=tfb.MaskedAutoregressiveFlow(made))
+
+  # Construct and fit model.
+  x_ = tfkl.Input(shape=(1,), dtype=tf.float32)
+  c_ = tfkl.Input(shape=(1,), dtype=tf.float32)
+  log_prob_ = distribution.log_prob(x_, bijector_kwargs={'conditional_input': c_})
+  model = tfk.Model([x_, c_], log_prob_)
+
+  model.compile(optimizer=tf.optimizers.Adam(),
+              loss=lambda _, log_prob: -log_prob)
+
+  batch_size = 25
+  model.fit(x=[x, c],
+          y=np.zeros((n, 0), dtype=np.float32),
+          batch_size=batch_size,
+          epochs=1,
+          steps_per_epoch=1,  # Usually `n // batch_size`.
+          shuffle=True,
+          verbose=True)
+
+  # Use the fitted distribution to sample condition on c = 1
+  distribution.sample((3,), bijector_kwargs={'conditional_input': np.ones((3, 1))})
```

Comment:
There is an important point to mention here: conditional_input must broadcast with shapes like: sample_shape + conditional_event_shape.